lib/rules/selector-max-compound-selectors/__tests__/index.mjs

```diff
@@ -384,3 +384,116 @@ testRule({
 		},
 	],
 });
+
+// Testing ignoreSelectors
+testRule({
+	ruleName,
+	config: [2, { ignoreSelectors: ['::v-deep', '/-moz-.*/', /-screen$/, '.ignored'] }],
+
+	accept: [
+		{
+			code: '::v-deep .foo .bar {}',
+			description: 'ignored selector at the beginning',
+		},
+		{
+			code: '.foo ::v-deep .bar {}',
+			description: 'ignored selector in the middle',
+		},
+		{
+			code: '.foo ::v-deep > .bar {}',
+			description: 'ignored selector in the middle followed by a combinator',
+		},
+		{
+			code: '.foo::v-deep .bar {}',
+			description: 'ignored selector as pseudo-element',
+		},
+		{
+			code: '.foo::v-deep > .bar {}',
+			description: 'ignored selector as pseudo-element followed by a combinator',
+		},
+		{
+			code: '.foo > ::v-deep .bar {}',
+			description: 'ignored selector preceded by a combinator',
+		},
+		{
+			code: '.foo input::-moz-placeholder {}',
+			description: 'selector ignored by string regex',
+		},
+		{
+			code: '.foo :-webkit-full-screen a {}',
+			description: 'selector ignored by regex',
+		},
+		{
+			code: '.foo::v-deep :-webkit-full-screen a ::-moz-placeholder {}',
+			description: 'multiple ignored selectors',
+		},
+		{
+			code: '.foo { & ::v-deep > .bar {} }',
+			description: 'nested ignored selector',
+		},
+		{
+			code: '.foo .bar > .ignored.ignored {}',
+			description: 'concatenated class selectors ignored one by one',
+		},
+		{
+			code: '.foo .bar > .ignored .ignored {}',
+			description: 'doubled class selectors ignored',
+		},
+	],
+
+	reject: [
+		{
+			code: '.foo .bar ::v-deep .baz {}',
+			description: 'ignored selector in the middle',
+			message: messages.expected('.foo .bar ::v-deep .baz', 2),
+			line: 1,
+			column: 1,
+			endLine: 1,
+			endColumn: 24,
+		},
+		{
+			code: '.foo .bar::v-deep .baz {}',
+			description: 'ignored selector as pseudo-element',
+			message: messages.expected('.foo .bar::v-deep .baz', 2),
+			line: 1,
+			column: 1,
+			endLine: 1,
+			endColumn: 23,
+		},
+		{
+			code: '.foo { & ::v-deep > .bar .baz {} }',
+			description: 'nested ignored selector',
+			message: messages.expected('.foo ::v-deep > .bar .baz', 2),
+			line: 1,
+			column: 8,
+			endLine: 1,
+			endColumn: 33,
+		},
+	],
+});
+
+testRule({
+	ruleName,
+	config: [2, { ignoreSelectors: [':not'] }],
+
+	accept: [
+		{
+			code: '.foo .bar:not(.baz) {}',
+		},
+		{
+			code: 'p a :not(.foo .bar) {}',
+		},
+	],
+
+	reject: [
+		{
+			code: 'p a :not(.foo .bar .baz) {}',
```

Comment:
minor
non-blocking

We could also check that `,` is supported.
i.e. `p a :not(.foo .bar, .foo .baz .qux) {}`

Reply:
Good catch. Added via 411ed1e9e